EIPS/eip-5247.md

```diff
@@ -0,0 +1,82 @@
+---
+eip: 5247
+title: Smart Proposal
+description: An interface of "proposals" that is submitted to, recorded on and possibly executed and enforced onchain.
+author: Zainan Victor Zhou (@xinbenlv)
+discussions-to: https://ethereum-magicians.org/t/erc-5247-executable-proposal-standard/9938
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-13
+requires: 7
+---
+
+## Abstract
+This EIP presents an interface for smart proposals an interface of "proposals" that is submitted to, recorded on and possibly executed and enforced onchain.
+When a smart proposal is on-chain executable, they contains information of smart-contract calls including addresses, ether values, calldatas and other metadata information.
+
+## Motivation
+Increasingly in use-cases we found it useful and sometimes necessary of separating the "what is to be executed" with "now execute it".
+
+A typical use case is in a Decentralized Autonomous Organization, aka DAO, a proposer will start a smart proposal, advocate it, let people endorse and vote (E.g. as [EIP-1202  Voting Standard](./eip-1202.md)) for it, and later when conesensus amongst community members are formed, the execution comes.
+
+A second typical use-case is that a person could have someone who they trust, such as a delegator, trustee, or an attorney-in-fact, or any bilateral collaboration format, where a smart proposal will be first composed, discussed, approved in some way, and then put into execution.
+
+A third typical use-case is that a person could make an "offer" to a second person, with or without strings attached. The smart proposal can be presented as a offer and the second person can execute it when they accept this proposal.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+
+interface IERC5247SmartProposal {
+    event ProposalCreated(
+        uint256 proposalId,
+        string proposalUri,
+        uint8[] optionIds,
+        address proposer,
+        address[][] targets,
+        uint256[][] values,
+        string[][] signatures,
+        bytes[][] calldatas,
+        uint256 startBlock,
+        uint256 endBlock
+    );
+    event ProposalExecuted(uint256 proposalId);
+    // TODO: add Proposal Cancel/Edit/Withdraw Event and Functions?
+
+    // TODO: decide whether we require generating ProposalId in the method or not?
+    // TODO: if require generating ProposalId internally, can it be incremental hash-generated?
+    // TODO: what if proposal need to demonstrate sufficient support? How to input quorum?
```

Comment:
Can I keep them as draft and as highlight these design decision to invoke discussion? I will guarantee to remove/resolve these TODOs before moving to review

Reply:
@SamWilsn would this be okay? I'm inclined to say yes but I'd like to have another opinion on this.

Reply:
I generally allow TODOs up until Review stage.  The original intent of Draft was that the barrier to entry for a draft should be very low, just enough to get you a number and a place people could start to submit PRs against.  We have drifted a bit away from that since then and also started requiring that people structure the file properly, but I don't think we should go so far as to require a draft EIP actually be finished.